diff --git a/scripts/setup_irods.py b/scripts/setup_irods.py
index 618f295c3..cfed8516c 100755
--- a/scripts/setup_irods.py
+++ b/scripts/setup_irods.py
@@ -375,6 +375,37 @@ def setup_client_environment(irods_config):
         input_filter=irods.lib.character_count_filter(minimum=3, maximum=maximum_password_length, field='Admin password'),
         echo=False)
 
+    irods_config.server_config['client_server_policy'] = irods.lib.prompt(
+        'Client server policy [CS_NEG_REQUIRE, CS_....TODO',
+        input_filter=irods.lib.character_count_filter(minimum=4, field='Client server policy'),
+        echo=True)
+
+    irods_config.server_config['ssl_settings'] = dict()
+    irods_config.server_config['ssl_settings']['certificate_chain_file'] = irods.lib.prompt(
+        '[SSL] Path to certificate chain file',
+        input_filter=irods.lib.character_count_filter(minimum=3, field='Path to certificate chain file'),
+        echo=True)
+
+    irods_config.server_config['ssl_settings']['certificate_key_file'] = irods.lib.prompt(
+        '[SSL] Path to certificate key file',
+        input_filter=irods.lib.character_count_filter(minimum=3, field='Path to certificate key file'),
+        echo=True)
+
+    irods_config.server_config['ssl_settings']['ca_certificate_file'] = irods.lib.prompt(
+        '[SSL] Path to CA certificate file',
+        input_filter=irods.lib.character_count_filter(minimum=3, field='Path to CA certificate file'),
+        echo=True)
+
+    irods_config.server_config['ssl_settings']['dh_params_file'] = irods.lib.prompt(
+        '[SSL] Path to Diffie-Hellman parameters file',
+        input_filter=irods.lib.character_count_filter(minimum=3, field='Path to Diffie-Hellman parameters file'),
+        echo=True)
+
+    irods_config.server_config['ssl_settings']['verify_server'] = irods.lib.prompt(
+        '[SSL] Verify server [none, cert, hostname]',
+        input_filter=irods.lib.character_count_filter(minimum=3, field='[SSL] Verify server [none, cert, hostname]'),
+        echo=True)
+
     print('\n', end='')
 
     service_account_dict = {
@@ -388,7 +419,7 @@ def setup_client_environment(irods_config):
             'irods_user_name': irods_config.server_config['zone_user'],
             'irods_zone_name': irods_config.server_config['zone_name'],
             'irods_client_server_negotiation': 'request_server_negotiation',
-            'irods_client_server_policy': 'CS_NEG_REFUSE',
+            'irods_client_server_policy': irods_config.server_config['client_server_policy'],
             'irods_encryption_key_size': 32,
             'irods_encryption_salt_size': 8,
             'irods_encryption_num_hash_rounds': 16,
@@ -403,6 +434,11 @@ def setup_client_environment(irods_config):
             'irods_default_number_of_transfer_threads': irods_config.server_config['advanced_settings']['default_number_of_transfer_threads'],
             'irods_transfer_buffer_size_for_parallel_transfer_in_megabytes': irods_config.server_config['advanced_settings']['transfer_buffer_size_for_parallel_transfer_in_megabytes'],
             'irods_connection_pool_refresh_time_in_seconds': 300,
+            'irods_ssl_certificate_chain_file': irods_config.server_config['ssl_settings']['certificate_chain_file'],
+            'irods_ssl_certificate_key_file': irods_config.server_config['ssl_settings']['certificate_key_file'],
+            'irods_ssl_ca_certificate_file': irods_config.server_config['ssl_settings']['ca_certificate_file'],
+            'irods_ssl_dh_params_file': irods_config.server_config['ssl_settings']['dh_params_file'],
+            'irods_ssl_verify_server': irods_config.server_config['ssl_settings']['verify_server'],
         }
     if not os.path.exists(os.path.dirname(irods_config.client_environment_path)):
         os.makedirs(os.path.dirname(irods_config.client_environment_path), mode=0o700)
