diff --git a/scripts/setup_irods.py b/scripts/setup_irods.py
index 352918726..a744703b5 100755
--- a/scripts/setup_irods.py
+++ b/scripts/setup_irods.py
@@ -120,14 +120,21 @@ def setup_server(irods_config, json_configuration_file=None):
         default_resource_directory = get_and_create_default_vault(irods_config)
 
     if irods_config.is_catalog:
-        l.info(irods.lib.get_header('Setting up the database'))
-        database_interface.setup_catalog(irods_config, default_resource_directory=default_resource_directory)
+        l.info(irods.lib.get_header('PATCHED setup_irods.py for development: Looked like an iCAT DB was already set up. So we skip setup_catalog().'))
+        #l.info(irods.lib.get_header('Setting up the database'))
+        #database_interface.setup_catalog(irods_config, default_resource_directory=default_resource_directory)
 
     l.info(irods.lib.get_header('Starting iRODS...'))
     IrodsController(irods_config).start()
 
     if irods_config.is_resource:
-        irods.lib.execute_command(['iadmin', 'mkresc', irods_config.server_config['default_resource_name'], 'unixfilesystem', ':'.join([irods.lib.get_hostname(), default_resource_directory]), ''])
+        lr_out = irods.lib.execute_command(['iadmin', 'lr', irods_config.server_config['default_resource_name']])[0]
+        if not (lr_out.find("No rows found") == 0):
+            l.info(irods.lib.get_header('PATCHED setup_irods.py for development. Looks like resource was already registered. Skipping mkresc.'))
+        else:
+            l.info(irods.lib.get_header('PATCHED setup_irods.py for development. Does not look like resource was already registered! Will mkresc'))
+            # Original line from setup_irods.py:
+            irods.lib.execute_command(['iadmin', 'mkresc', irods_config.server_config['default_resource_name'], 'unixfilesystem', ':'.join([irods.lib.get_hostname(), default_resource_directory]), ''])
 
     test_put(irods_config)
 
