--- /var/lib/irods/scripts/setup_irods.py	2024-05-31 10:22:27.120880851 +0200
+++ setup_modified.py	2024-05-31 10:21:52.080199164 +0200
@@ -141,7 +141,14 @@
     # prompt, the local storage resource should not be created. This is indicated by default_resource_directory being
     # None, as returned by setup_storage().
     if irods_config.is_consumer and default_resource_directory is not None:
-        irods.lib.execute_command(['iadmin', 'mkresc', default_resource_name, 'unixfilesystem', ':'.join([irods.lib.get_hostname(), default_resource_directory]), ''])
+        #irods.lib.execute_command(['iadmin', 'mkresc', default_resource_name, 'unixfilesystem', ':'.join([irods.lib.get_hostname(), default_resource_directory]), ''])
+        lr_out = irods.lib.execute_command(['iadmin', 'lr', irods_config.server_config['default_resource_name']])[0]
+        if not (lr_out.find("No rows found") == 0):
+            l.info(irods.lib.get_header('PATCHED setup_irods.py for development. Looks like resource was already registered. Skipping mkresc.'))
+        else:
+            l.info(irods.lib.get_header('PATCHED setup_irods.py for development. Does not look like resource was already registered! Will mkresc'))
+            # Original line from setup_irods.py:
+            irods.lib.execute_command(['iadmin', 'mkresc', default_resource_name, 'unixfilesystem', ':'.join([irods.lib.get_hostname(), default_resource_directory]), ''])
 
     # update core.re with default resource
     core_re_path = os.path.join(irods_config.core_re_directory, 'core.re')
